make_checkpoint: rewrite code for every100th iterations (#3983)
iterative/dvc.org: content/docs/api-reference/make_checkpoint.md

```yaml
stages:
  train:
    cmd: python train.py
    outs:
      - model:
          checkpoint: true
```

The code in `train.py` will train a model upto certain epochs. At every 100 iterations,
it saves the model to `model`, evaluates it and makes a checkpoint for [DVC experiments]:

[dvc experiments]: /doc/user-guide/experiment-management#experiments

```py
from dvc.api import make_checkpoint

for epoch in range(epochs):

>     train(model, x_train, y_train)

Author: I think `model`/`x_train`/`y_train`/`x_test`/`y_test` are common namings that we don't have to show them in the code.

```yaml
stages:
  train:
    cmd: python train.py
    outs:
      - model:
          checkpoint: true
```

> The code in `train.py` will train a model upto certain epochs. At every 100
> iterations, it saves the model to `model`, evaluates it and makes a checkpoint

Reviewer: ```suggestion
The code in `train.py` will train a model up to a number of epochs. Every 100
iterations, it saves the `model`, evaluates it, and makes a checkpoint
```

for [DVC experiments]:

[dvc experiments]: /doc/user-guide/experiment-management#experiments

```py

> from dvc.api import make_checkpoint

Author: I wonder why we named it `make_checkpoint()` instead of just `checkpoint()`.

Reviewer 2: I used `make_checkpoint` because `checkpoint` is a noun in this context and IMO method names should almost always be verbs.

> > DVC checkpoints to behave as expected.

Author: There may be other errors too, so i am not sure whether we should show example of catching `KeyboardInterrupt`.

Reviewer: I think the code sample was taken from an actual checkpoints example repo which is a demo meant to run continuously until you Ctrl + C.

Reviewer 3: See https://github.com/iterative/dvc-checkpoints-mnist/pull/2

Author: @dberenbaum, I think that could be and should be explained separately. I'd keep it simple for `every 100th iteration` example.
